nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkStatus.java

```diff
@@ -35,69 +34,76 @@
     private static final long serialVersionUID = 1L;
 
     /**
-     * Possible states of bulk operation.
+     * Possible states of bulk execution.
      */
     public enum State {
-        SCHEDULED, BUILDING, COMPLETED
+        /** The {@link BulkCommand} has been submitted to the system. */
+        SCHEDULED,
+
+        /** System is currently scrolling the database. */
+        BUILDING,
+
+        /** System has finished to scroll. */
+        COMPLETED
```

Comment:
Don't we need an additional state to denote that scrolling is in progress ? SCHEDULED -> SCROLLING_RUNNING -> RUNNING -> COMPLETED

Reply:
Because `SCROLLING_RUNNING` and `RUNNING` could reflect state of bulk at the same moment. I think we should use two states, one for scroller and another for actions.
We were wondering with @funshodavid, how to detect `RUNNING`, bulk actions could send to status stream an event to set this status (and eventually begin date) if it's not yet set. But this could mean a lot of requests to K/V, wdyt ?